aten/src/ATen/native/BackendSelectRegister.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <c10/core/TensorOptions.h>
+
+// TODO:
+// 1) add all factory functions
+// 2) autogen this
+
+namespace at {
+namespace native {
+  Tensor empty_backend(IntArrayRef size, const TensorOptions & options, c10::optional<MemoryFormat> memory_format) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::empty", "memory_format");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, const TensorOptions &, c10::optional<MemoryFormat>>(key, size, options, memory_format);
+  }
+
+  Tensor empty_strided_backend(IntArrayRef size, IntArrayRef stride, const TensorOptions & options) {
```

Comment:
this indentation (inside the namespace) doesn't follow other examples I checked in this folder.